programs/drift/src/controller/orders.rs

```diff
@@ -1768,60 +1768,61 @@ pub fn fulfill_perp_order_with_match(
     };
 
     let mut total_quote_asset_amount = 0_u64;
-    let base_asset_amount_left_to_fill = if amm_wants_to_make && market.amm.amm_jit_is_active() {
-        let jit_base_asset_amount = if !taker.orders[taker_order_index].has_limit_price(slot)?
-            && maker_base_asset_amount < taker_base_asset_amount
-        {
-            0
-        } else {
-            crate::math::amm_jit::calculate_jit_base_asset_amount(
-                market,
-                base_asset_amount,
-                maker_price,
-                valid_oracle_price,
-                taker_direction,
-            )?
-        };
-
-        if jit_base_asset_amount > 0 {
-            let (base_asset_amount_filled_by_amm, quote_asset_amount_filled_by_amm) =
-                fulfill_perp_order_with_amm(
-                    taker,
-                    taker_stats,
-                    taker_order_index,
+    let taker_base_asset_amount_left_to_fill =
```

Comment:
i think we might be able to avoid all of this nesting if instead of calculating `taker_base_asset_amount_left_to_fill`, we just pull the `base_asset_amount_unfilled` off of the taker order after the jit block

Reply:
yeah great point